circuitbreaker/circuitbreaker.go

```diff
@@ -10,14 +10,30 @@ import (
 	"github.com/slok/goresilience/metrics"
 )
 
-type state string
+type state uint
 
 const (
-	stateOpen     state = "open"
-	stateHalfOpen state = "halfopen"
-	stateClosed   state = "closed"
+	stateNew   state = iota
+	stateOpen
+	stateHalfOpen
+	stateClosed
 )
 
+var stateStrings = []string {
+	"new",
+	"open",
+	"halfopen",
+	"closed",
+}
+
+func (st state) label () string {
+	return stateStrings[st]
+}
+
+func (st state) condition () int {
```

Comment:
I think that we could remove the integer usage for `state` and use only the type and the string representation of it. It would simplify the usage and we remove code.

Reply:
I agree, will refactor accordingly. (Am I right in that you're happy with the use of ```iota```?

Reply:
The `iota` is perfect